tao_compiler/mlir/xla/ral/context/mkldnn/ideep/ideep/operators/matmul.hpp

```diff
@@ -53,10 +53,10 @@ struct matmul_forward : public dnnl::matmul,
       data_type x_dtype = data_type::f32,
       const engine& aengine = engine::cpu_engine()) {
     auto ndims = weights_dims.size();
-    auto x_dims = weights_dims;
+    dims x_dims = weights_dims;
```

Comment:
just curious, why can not use `auto` ?

Reply:
Two reasons：
- using `auto` in line 59 is a bug (auto-detected type is not dims)
- using `auto` in line 57 is ok, while we follow community best practice to avoid using auto as long as it will not increase two much typings.